### PR TITLE
Creating CRUD controllers for memes

### LINKED ISSUE #6: CRUD controllers for memes
- [x] Create meme
- [x] Get meme / Get All memes
- [x] Update meme
- [x] Remove meme

### LINKED ISSUE #7: Firestore and Firebase Storage setup
- [x] Creating collection for users, memes and likes
- [x] Connecting with server 
- [x] Uploading media files to Cloud Storage
- [x] Integration

### PR DESCRIPTION
**Pull request by vketrys**
- [x] Creating CRUD controllers
- [x] Using Firestore as database 
- [x] Using Firebase Storage as Cloud Storage for files
- [x] Implement media uploading

### CODE REVIEW

File: src/constants/URL.ts [hunk 1]
```diff
@@ -5,7 +5,11 @@ export const URL = {
 		SIGNUP: '/signup',
 	},
 	USERS: {
-		GET: '/users',
+		SIMPLE: '/users',
```
Comment (esnts) [inline src/constants/URL.ts]: What is SIMPLE? 
image

File: src/controllers/authControllers.ts [hunk 2]
```diff
@@ -9,11 +9,15 @@ import app from '../config/firebase.js';
...
+import { db } from '../index.js';
+import Collections from '../constants/collections.js';
 
 export const signup = async(req: Request, res: Response): Promise<Response> => {
 	try {
 		const { displayName, email, password } = req.body;
-		const role = Roles.user;
+
+		//temporary
+		const role = email === 'admin@gmail.com' ? Roles.admin : Roles.user;
```
Comment (esnts) [inline src/controllers/authControllers.ts]: U can move it to en file

File: src/controllers/memeControllers.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	}
+
+	const fileType = file.originalname.split('.').at(-1);
+
+	try {
+		const memeRef = await db.collection(Collections.memes).add({
+			text,
+			isPublished: false,
+			likes: 0,
+			userId: uid,
```
Comment (esnts) [inline src/controllers/memeControllers.ts:28]: 👍🏻

File: src/controllers/memeControllers.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+		if (likeSnap.docs.length === 0) {
+			await likeRef.add({
+				uid,
+				memeId,
+			});
+
+			await memeRef.update({ likes: increment });
+		} else {
+
+			likeSnap.docs.forEach((doc) => doc.ref.delete());
```
Comment (esnts) [inline src/controllers/memeControllers.ts:89]: Mb it is better to do nothing if user already liked this one?

File: src/constants/URL.ts (hunk 1 above)
Comment (vketrys) [reply src/constants/URL.ts]: not that simple, it means without params

File: src/controllers/authControllers.ts (hunk 2 above)
Comment (vketrys) [reply src/controllers/authControllers.ts]: but why if this is tempo ?

File: src/controllers/authControllers.ts [hunk 5]
```diff
@@ -4,16 +4,23 @@ import {
...
+import Collections from '../constants/collections.js';
+
+dotenv.config();
 
 export const signup = async(req: Request, res: Response): Promise<Response> => {
 	try {
 		const { displayName, email, password } = req.body;
-		const role = Roles.user;
+
+		//temporary
```
Comment (slamser56) [inline src/controllers/authControllers.ts]: Would be better if you add TODO comment with explanation what you or someone will need to do here

Comment (vketrys) [reply src/controllers/authControllers.ts]: ok, got it !

**Review by esnts — APPROVED** [review]

**Pull request merged**